### PR TITLE
NOISSUE - add edge docs

### PR DESCRIPTION
**Pull request by mteodor**

### CODE REVIEW

**Review by drasko — CHANGES_REQUESTED** [review]
Regarding diagram:
- Agent comm with NATS is bidirectional
- Export has Redis Streams for local buffering
- Add some colors and write what is the Gateway (in the box)

File: docs/agent.md [hunk 1]
```diff
@@ -1,8 +1,6 @@
 Agent is application that is used to manage gateways that are connected to Mainflux. It provides a way to send commands to gateway and receive response via mqtt.
-There are two type of channels used for **Agent** `data` and `control`.
-Over the `control` we are sending commands and receiving response from commands.
-Data collected from sensors connected to gateway are being sent over `data` channel.
-Agent is able to configure itself provided that [bootstrap server](./bootstrap.md) is running, it will retrieve configuration from bootstrap server provided few arguments - `external_id` and `external_key` see [bootstraping](./bootstrap.md/#bootstraping).
+There are two type of channels used for **Agent** `data` and `control`. Over the `control` we are sending commands and receiving response from commands.
```
Comment (dborovcanin) [inline docs/agent.md]: There are two types (plural) of channels used for **Agent**: `data` and `control`.

File: docs/edge-deployment.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
+# Edge deployment
+
+Following are steps that are an example usage of Mainflux components to connect edge with cloud.
```
Comment (dborovcanin) [inline docs/edge-deployment.md]: The following...

File: docs/edge-deployment.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+We will start Mainflux in the cloud with additional services [Bootstrap](bootstrap) and [Provision](provision).
+Using [Bootstrap](bootstrap) and [Provision](provision) we will create configuration for use in gateway deployment.
```
Comment (dborovcanin) [inline docs/edge-deployment.md]: ... a configuration for use...

File: docs/edge-deployment.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    }
+  ],
+  "whitelisted": {
+    "88529fb2-6c1e-4b60-b9ab-73b5d89f7404": true
+  }
+}
+```
+
+Parameters <external_id> and <external_key> are representing the gateway. `Provision` will use them to create a bootstrap configuration that will make a relation with Mainflux entities used for connection, authentication and authorization `thing` and `channel`.
+These paramaters will be used by `Agent` service on the gateway to retrieve that information and establish a connection with the cloud.
```
Comment (dborovcanin) [inline docs/edge-deployment.md]: ...parameters... by the `Agent` service...

File: docs/edge.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
+## Edge 
+
+Mainflux IoT platform provides services for supporting management of devices in the edge. Typicaly, IoT solution includes devices (sensors/actuators) deployed in far edge and connected through some proxy gateway.
```
Comment (dborovcanin) [inline docs/edge.md]: ...on the edge. Tipically, ...

File: docs/edge.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
+## Edge 
+
+Mainflux IoT platform provides services for supporting management of devices in the edge. Typicaly, IoT solution includes devices (sensors/actuators) deployed in far edge and connected through some proxy gateway. 
+Although most devices could be connected to the Mainflux directly using gateways provides for data transmission and routing eases a process of set up. Also gateways can provide additional data processing, filtering and storage.
```
Comment (dborovcanin) [inline docs/edge.md]: ...a process of setting up.

File: docs/export.md [hunk 7]
```diff
@@ -1,55 +1,152 @@
-Bridging between two mainflux instances can be done with [export](https://github.com/mainflux/export) service is build to bridge between two Mainflux instances. For example we can run Mainflux on the gateway and
-we want to export the data to cloud instance of Mainflux as well.
-Export service is listening to the NATS and forwards payload to the specified MQTT channel.
+# Export
+Mainflux Export service can send message from one Mainflux cloud to another via MQTT, or it can send messages from edge gateway to Mainflux Cloud.
+Export service is subscribed to local message bus and connected to MQTT broker in the cloud.  
+Messages collected on local message bus are redirected to the cloud.
+When connection is lost, messages from local bus are stored into `Redis` stream. Upon connection reestablishment `Export` service consumes messages from `Redis` stream and sends it to the Mainflux cloud.
```
Comment (dborovcanin) [inline docs/export.md]: ... from the local bus...

**Review by dborovcanin — CHANGES_REQUESTED** [review]

File: docs/edge.md [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+
+Mainflux IoT platform provides services for supporting management of devices on the edge. Typically, IoT solution includes devices (sensors/actuators) deployed in far edge and connected through some proxy gateway. 
+Although most devices could be connected to the Mainflux directly using gateways provides for data transmission and routing eases a process of setting up. Also gateways can provide additional data processing, filtering and storage.
+
+Services that can be used on gateway to enable data and control plane for edge:
+
+* [Agent][agent]
+* [Export][export]
+
+![Edge](img/edge/edge.png)
```
Comment (drasko) [inline docs/edge.md:11]: Here you kept nice link, however up you modified to rawcontent. Why?

File: docs/bootstrap.md [hunk 9]
```diff
@@ -11,7 +11,7 @@ Mainflux platform supports bootstrapping process, but some of the preconditions
 
 Bootstrapping procedure is the following:
 
-![Configure device](img/bootstrap/1.png)
+![Configure device](https://raw.githubusercontent.com/mainflux/docs/master/docs/img/bootstrap/1.png)
```
Comment (drasko) [inline docs/bootstrap.md]: Do we need this? It really can not work with relative link?

**Review by drasko — CHANGES_REQUESTED** [review]

File: docs/edge.md (hunk 8 above)
Comment (mteodor) [reply docs/edge.md:11]: you mean like this `![Edge][edge]`? it didnt work.

Comment (drasko) [reply docs/edge.md:11]: No, this is good. But you changed the other one. Revert is good, so I resolve this one.

**Review by drasko — APPROVED** [review]
LGTM

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**